### ISSUE TITLE
Use fn:parse-uri to check whether a filepath is relative or absolute

### ISSUE THREAD

**Issue by fsteimke** [qt4cg/qtspecs#1176]
I have a question about the new function `fn:parse-uri()`. A common use case is to check whether a file path is absolute or relative. For example, I want to check whether the file path `images/img1.png` is relative and can therefore be converted to an absolute file path using `resolve-uri()`. Or I want to check whether `$base` is absolute and can therefore be used as the second argument in `resolve-uri()`.

How would I use a `uri-structure-record` map determined as a result of` fn:parse-uri ()` to decide whether it is a relative or absolute file path?  
Greetings, Frank

**Comment by ChristianGruen** [comment]
You can take advantage of the `hierarchical` property: 

```xquery
parse-uri('a')?hierarchical  →  false
parse-uri('/a')?hierarchical  →  true
parse-uri('c:/a')?hierarchical  →  true
parse-uri('file://a')?hierarchical  →  true
```

The property may be absent if the input has no path:

```xquery
parse-uri('?a')?hierarchical  →  ()
parse-uri('#a')?hierarchical  →  ()
```

**Comment by michaelhkay** [comment]
I'm not sure whether that really answers the question. 

In fact there seem to be two questions. The first part is whether file paths are absolute or relative. The second part ("whether $base is relative") is about whether a URI is absolute or relative. 

`file://a` is an absolute URI, while `/a` as a filepath is absolute, but as a URI is relative.

**Comment by ChristianGruen** [comment]
> The second part ("whether $base is relative") is about whether a URI is absolute or relative.

True; I missed that one. I assume the the `scheme` property would additionally need to be taken into account. Maybe there is a better way, I don’t know?

**Comment by ndw** [comment]
Whether or not `hierarchical` is true depends only on the implementation's understanding of the scheme.  I think the answer is, if `hierarchical` is true *and* `path-segments[1] = ""` then the URI was absolute.

**Comment by fsteimke** [comment]
I had hoped for a simple answer like this.

If it holds true for every argument that can be successfully analyzed by the new  `parse-uri()` function  (URI as well as file paths),  i would appreciate a note about it in the upcoming specification.

**Comment by ndw** [comment]
I have outstanding actions in that area. I'll try to clarify it. I should say, you've almost persuaded me to add an `absolute` property for this case. It seems quite likely to be a common case.

**Comment by ndw** [comment]
Consensus at the PRG f2f, add the property.

**Comment by michaelhkay** [comment]
Am I right in thinking we can close this as completed?

**Comment by ndw** [comment]
At [meeting 106](https://qt4cg.org/meeting/minutes/2025/01-21.html) the CG agreed to close this issue without further action.
